realm/realm-library/src/main/java/io/realm/RealmResults.java

```diff
@@ -831,15 +824,17 @@ public void removeChangeListeners() {
      * Notifies all registered listeners.
      */
     void notifyChangeListeners() {
-        // table might be null (if the async query didn't complete
-        // but we have already registered listeners for it)
-        if (pendingQuery != null && !isCompleted) return;
-
-        long version = table.sync();
-        if (currentTableViewVersion != version) {
-            currentTableViewVersion = version;
-            for (RealmChangeListener listener : listeners) {
-                listener.onChange();
+        if (listeners != null && !listeners.isEmpty()) {
+            // table might be null (if the async query didn't complete
+            // but we have already registered listeners for it)
+            if (pendingQuery != null && !isCompleted) return;
+
+            long version = table.sync();
```

Comment:
Should we check if `table` is a `TableView`? If not, an exception will be thrown.